addons/test_mail/tests/test_message_post.py

```diff
@@ -239,16 +239,26 @@ def test_post_notify(self):
         new_notification = self.test_record.message_notify(
             subject='This should be a subject',
             body='<p>You have received a notification</p>',
-            partner_ids=[self.partner_1.id, self.user_employee.partner_id.id],
+            partner_ids=[self.partner_1.id, self.partner_admin.id, self.user_employee.partner_id.id],
         )
 
         self.assertEqual(new_notification.subtype_id, self.env.ref('mail.mt_note'))
         self.assertEqual(new_notification.message_type, 'user_notification')
         self.assertEqual(new_notification.body, '<p>You have received a notification</p>')
         self.assertEqual(new_notification.author_id, self.env.user.partner_id)
         self.assertEqual(new_notification.email_from, formataddr((self.env.user.name, self.env.user.email)))
-        self.assertEqual(new_notification.notified_partner_ids, self.partner_1 | self.user_employee.partner_id)
+        self.assertEqual(new_notification.notified_partner_ids, self.partner_1 | self.user_employee.partner_id | self.partner_admin)
         self.assertNotIn(new_notification, self.test_record.message_ids)
+
+        admin_mail = [x for x in self._mails if self.partner_admin.name in x.get('email_to')[0]]
+        admin_mail = admin_mail[0].get('body') if admin_mail else None
+        partner_mail = [x for x in self._mails if self.partner_1.name in x.get('email_to')[0]]
+        partner_mail = partner_mail[0].get('body') if partner_mail else None
+        admin_access_link = admin_mail[admin_mail.index('model='):admin_mail.index('/>') - 1] if 'model=' in admin_mail else None
+
+        self.assertNotIn('/mail/view?model=', partner_mail)
+        self.assertIn('model=%s' % self.test_record._name, admin_access_link)
+        self.assertIn('res_id=%d' % self.test_record.id, admin_access_link)
```

Comment:
Don't hesitate to add an explanation on your assertions (the third param) so the reader understands more easily what you're asserting